### PR TITLE
[wasm][debugger] Rework setBreakpointByUrl location handling

### PR DESCRIPTION
**Pull request by lewing**
- For column=0 requests only set and return the best match
- Merge js/html locations into the response so that they don't appear to be unbound
- Handle the possibility of multiple documents source files matching the regex but only use the best match from each

### CODE REVIEW

**Comment by lewing** [comment]
fixes some cases that would manifest like https://github.com/mono/mono/issues/19682

**Comment by lewing** [comment]
fixes https://github.com/mono/mono/issues/19689

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs [hunk 1]
```diff
@@ -765,17 +767,20 @@ async Task SetBreakpoint (SessionId sessionId, DebugStore store, BreakpointReque
 				return;
 			}
 
-			var locations = store.FindBreakpointLocations (req).ToList ();
+			var comparer = new SourceLocation.LocationComparer ();
+			// if column is specified the frontend wants the exact matches
+			// and will clear the bp if it isn't close enoug
```
Comment (lewing) [inline sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs:772]:
```suggestion
			// and will clear the bp if it isn't close enough
```

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs [hunk 2]
```diff
@@ -765,17 +767,20 @@ async Task SetBreakpoint (SessionId sessionId, DebugStore store, BreakpointReque
 				return;
 			}
 
-			var locations = store.FindBreakpointLocations (req).ToList ();
+			var comparer = new SourceLocation.LocationComparer ();
+			// if column is specified the frontend wants the exact matches
+			// and will clear the bp if it isn't close enough
+			var locations = store.FindBreakpointLocations (req)
+				.Distinct (comparer)
+				.Where (l => l.Line == req.Line && (req.Column == 0 || l.Column == req.Column))
```
Comment (radical) [inline sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs:775]: Is this with 1-based column numbers?

Comment (lewing) [reply sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs:775]: no, SourceLocation and BreakpointRequest are zero based like cdp

```
public SourceLocation (MethodInfo mi, SequencePoint sp)
		{
			this.id = mi.SourceId;
			this.line = sp.StartLine - 1;
			this.column = sp.StartColumn - 1;
			this.cliLoc = new CliLocation (mi, sp.Offset);
		}
```

Comment (radical) [reply sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs:775]: Then how do we differentiate between requests for exactly `column:0`, and inexact column (all breakpoints matching the line)?

Comment (lewing) [reply sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs:775]: it doesn't matter the point of the patch is that column:0 == first bp on line whatever the column.

**Comment by radical** [comment]
Two of the test failures are timeouts.

**Comment by lewing** [comment]
@monojenkins backport 3.2-wasm

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build Linux WebAssembly

**Pull request merged**